### PR TITLE
Add plugin support to weave script

### PR DESCRIPTION
**Pull request by awh**
Implements part of weaveworks/docker-plugin#36.

### CODE REVIEW

**Comment by awh** [comment]
Thinking about this some more - if we're going to do a prerelease build of weave with plugin support, should we base it off `1.2` instead of `master`? Thoughts please @bboreham @rade

**Comment by bboreham** [comment]
As the plugin is targeted for the 1.3 release we should prerelease off master not 1.2

**Comment by awh** [comment]
I didn't mean that we should commit the changes to the `1.2` branch - we could branch from `1.2`, make a prerelease from there and then merge that branch back into `master` afterwards. If we were for example worried about the large amount of relatively untested change that has gone onto `master` recently...

File: weave [hunk 1]
```diff
@@ -1781,8 +1811,7 @@ EOF
         fi
         collect_cidr_args "$@"
         shift $CIDR_ARG_COUNT
-        CONTAINER=$(docker run -e WEAVE_CIDR=none $DNS_ARGS -d "$@")
-        create_bridge
+        CONTAINER=$(docker run -e WEAVE_CIDR=none $DNS_ARGS -d "$@") create_bridge
```
Comment (bboreham) [inline weave]: Did you mean to merge these two lines?

File: weave [hunk 2]
```diff
@@ -1705,6 +1724,12 @@ EOF
         launch_proxy "$@"
         echo $PROXY_CONTAINER
         ;;
+    launch-plugin)
+        deprecation_warnings "$@"
```
Comment (bboreham) [inline weave]: Shouldn't need this for a brand-new feature

File: weave [hunk 3]
```diff
@@ -1760,6 +1785,11 @@ EOF
             echo "       Service: proxy"
             echo "       Address: $PROXY_ADDRS"
         fi
+        if [ -z "$SUB_STATUS" ] && check_running $PLUGIN_CONTAINER_NAME 2>/dev/null ; then
+            echo
+            echo "       Service: plugin"
+            echo "       Address: unix:///run/docker/plugins/weave.sock"
```
Comment (bboreham) [inline weave]: This information isn't very useful to the user, is it?  How about "Driver name: weave" ?

**Comment by bboreham** [comment]
Do you think we should check for minimum Docker 1.9 when using the plugin?

File: site/features.md [hunk 4]
```diff
@@ -88,6 +89,22 @@ Containers started in this way that subsequently restart, either by an
 explicit `docker restart` command or by Docker restart policy, are
 re-attached to the weave network by the weave Docker API proxy.
 
+### <a name="plugin"></a>Docker network plugin
+
+Alternatively, you can use weave as a Docker plugin. First you must
+configure the Docker daemon to use a KV store via the
```
Comment (bboreham) [inline site/features.md]: Docker don't use the term "KV store"; they do use "KV service" once.  Maybe just "configure the Docker daemon to use a cluster store as described here"

**Comment by awh** [comment]
> Do you think we should check for minimum Docker 1.9 when using the plugin?

This is a bit tricky, because under normal circumstances (e.g. sans `--local`) `launch-plugin` is happening inside the `weaveexec` container which has a fixed client version. Is it that important?

**Comment by bboreham** [comment]
I don't suppose the version check is too important; the plugin will just sit there with no-one talking to it, and the `docker network create` will fail.

**Pull request merged**